djtriggers/tests/test_models.py

```diff
@@ -0,0 +1,108 @@
+from datetime import timedelta
+from logging import WARNING
+
+import mock
+import pytest
+from django.test import override_settings
+from django.test.testcases import TestCase
+from django.utils import timezone
+from locking.models import NonBlockingLock
+
+from djtriggers.exceptions import ProcessLaterError, AlreadyProcessedError
+from djtriggers.loggers.base import TriggerLogger
+from djtriggers.models import Trigger
+from djtriggers.tests.factories.triggers import NormalDummyTriggerFactory, TimelyDummyTriggerFactory, \
+    FailingDummyTriggerFactory
+from djtriggers.tests.models import NormalDummyTrigger, FailingDummyTrigger, TimelyDummyTrigger
+
+
+class TriggerTest(TestCase):
+    """
+    Due to some bug in celery.contrib.methods.task_method testing the tash method needs be done in the following way:
+        Class.TaskMethod(class_instance)
+    """
+    def setUp(self):
+        pass
+
+    def test_handle_execution_success(self):
+        trigger = NormalDummyTriggerFactory()
+        trigger._handle_execution_success()
+
+        assert trigger.date_processed
+        assert timezone.now() - trigger.date_processed < timedelta(minutes=1)
+
+    @mock.patch('django_statsd.clients.statsd')
+    def test_test_handle_execution_success_use_statsd(self, mock_statsd):
+        trigger = NormalDummyTriggerFactory(process_after=timezone.now())
+        trigger._handle_execution_success(use_statsd=True)
+
+        assert trigger.date_processed
+        assert timezone.now() - trigger.date_processed < timedelta(minutes=1)
+        mock_statsd.incr.assert_called_with('triggers.%s.processed' % trigger.trigger_type)
+        mock_statsd.timing.assert_called_with('triggers.%s.process_delay_seconds' % trigger.trigger_type,
+                              (trigger.date_processed - trigger.process_after).total_seconds())
+
+    def test_handle_execution_failure(self):
+        exception = Exception()
```

Comment:
In the testcase without 'use_statsd' I don't need it, in the testcase with 'use_statsd' I need it to construct the message to assert.